drivers/bluetooth/hci/ipc.c

```diff
@@ -265,19 +266,30 @@ static int bt_ipc_send(const struct device *dev, struct net_buf *buf)
 		break;
 	default:
 		LOG_ERR("Unknown type %u", bt_buf_get_type(buf));
+		err = -ENOMSG;
 		goto done;
 	}
 	net_buf_push_u8(buf, pkt_indicator);
 
 	LOG_HEXDUMP_DBG(buf->data, buf->len, "Final HCI buffer:");
-	err = ipc_service_send(&data->hci_ept, buf->data, buf->len);
+
+	do {
+		err = ipc_service_send(&data->hci_ept, buf->data, buf->len);
+		if (err < 0) {
+			retries++;
+			k_msleep(CONFIG_BT_HCI_IPC_SEND_RETRY_DELAY_MS);
+		}
+	} while ((err == -ENOMEM) && (retries < CONFIG_BT_HCI_IPC_SEND_RETRY_COUNT));
```

Comment:
The Kconfig mentions that it's a generic retry, but it looks like it will only ever retry on `-ENOMEM`. I think there's a small misalignment between the Kconfig option and the implementation

Reply:
The -ENOMEM was chosen for the "when no memory / buffers are available." return value for ipc_serivice_send. This error usually occurs when the shared memory (or other medium) is full and the remote side hasn't processed the messages yet.

Reply:
Expecting retry on generic error would be strange.
I understand retry in this context happens only if IPC is busy (e.g., no buffers available). I expect this is what @dchat-nordic meant. If there is confusion it should indeed be reworded.

Reply:
@dchat-nordic I think the Kconfig help is still missing explicit description that the retry is only in case of out-of-memory errors

Reply:
I added more specific help to the Kconfig.